x/interchainstaking/keeper/hooks.go

```diff
@@ -23,6 +22,15 @@ func (k *Keeper) BeforeEpochStart(_ sdk.Context, _ string, _ int64) error {
 //
 // and re-queries icq for new zone info.
 func (k *Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	// every day
```

Comment:
change this to to triggering a ICA balance query against the deposit account, with a new callback.

callback should check for recent (epochDay.length) receipts and discount the amount, and then do a single delegation (with a special memo that is picked up by MsgSendHandler, that updates the completion time off all relevant receipts).

Reply:
[0b37d3e](https://github.com/ingenuity-build/quicksilver/pull/402/commits/0b37d3e1a9ae81799f18c502ce8cc4431bd5c4d2)